### PR TITLE
Add execution tests for AbstractFloat -> f16 assignment

### LINKED ISSUE #2618: wgsl: execution tests for assigning `AbstractFloat` to `f16`
For values precisely defined in f16, the result should be exact. Otherwise it should be correctly rounded.

### PR DESCRIPTION
**Pull request by zoddicus**
This adds in a stub implementation of F16Traits, with enough defined
to get correctlyRounded working.

It also adds in a unimplementeFoo feature for FPTraits, which is used
in FPAbtractTraits, since that code is untested/stubbed.

Fixes #2618

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by zoddicus** [comment]
This depends on https://github.com/gpuweb/cts/pull/2619, so the first diff is just a squashed version of that change.

**Comment by zoddicus** [comment]
@jzm-intel FYI, this PR introduces a stub implementation for the FP16Traits. I am OOO until Tuesday, so this probably won't land until then. If you have PRs that might conflict, feel free to proceed with them, and I will resolve conflicts on my side.

File: src/webgpu/util/floating_point.ts
Comment (ben-clayton) [inline src/webgpu/util/floating_point.ts:43]: übernit: I would have put `f16` next to `f32`.

**Review by ben-clayton — APPROVED** [review]

File: src/webgpu/util/floating_point.ts
Comment (zoddicus) [reply src/webgpu/util/floating_point.ts]: Done

**Pull request merged**
